python/ray/train/examples/jax_mnist_example.py

```diff
@@ -0,0 +1,220 @@
+# This example showcases how to use Jax (pmap) with Ray Train.
+# Original code: (without pmap)
+# https://github.com/google/flax/blob/main/examples/mnist/train.py
+import argparse
+from typing import Dict
+from ray.air import session
+
+import functools
+import time
+
+from ray.air.config import ScalingConfig
+from ray.train.jax import JaxTrainer
+import einops
+import numpy as np
+import optax
+import tensorflow_datasets as tfds
+
+
+def get_datasets():
+    import jax
+
+    """Load MNIST train and test datasets into memory."""
+    # shard the dataset
+    def shard_fn(x):
+        # shard the dataset for each node (each process)
+        return einops.rearrange(x, "(d l) ... -> d l ...", d=jax.process_count())[
+            jax.process_index()
+        ]
+
+    # Hide any GPUs from TensorFlow. Otherwise TF might reserve memory and make
+    # it unavailable to JAX.
+    import tensorflow as tf
+
+    tf.config.experimental.set_visible_devices([], "GPU")
+
+    ds_builder = tfds.builder("mnist")
+    ds_builder.download_and_prepare()
+    test_ds = tfds.as_numpy(ds_builder.as_dataset(split="test", batch_size=-1))
+    train_ds = tfds.as_numpy(ds_builder.as_dataset(split="train", batch_size=-1))
+
+    # TODO: warning about the sharding dimension
+    # train_ds["image"] = train_ds["image"][
+    #     : len(train_ds["image"]) // jax.device_count() * jax.device_count()
+    # ]
+    # train_ds["label"] = train_ds["label"][
+    #     : len(train_ds["label"]) // jax.device_count() * jax.device_count()
+    # ]
+    # test_ds["image"] = test_ds["image"][
+    #     : len(test_ds["image"]) // jax.device_count() * jax.device_count()
+    # ]
+    # test_ds["label"] = test_ds["label"][
+    #     : len(test_ds["label"]) // jax.device_count() * jax.device_count()
+    # ]
+
+    train_ds["image"] = np.float32(shard_fn(train_ds["image"])) / 255.0
+    test_ds["image"] = np.float32(shard_fn(test_ds["image"])) / 255.0
+    train_ds["label"] = np.int32(shard_fn(train_ds["label"]))
+    test_ds["label"] = np.int32(shard_fn(test_ds["label"]))
```

Comment:
Just create new dictionaries for the results?

Reply:
sounds good to me